### PR TITLE
[PRM-328] - added health appeal pages for no hospital stay

### PR DESCRIPTION
**Pull request by BackRowOla**
Added Did health issue include hospital stay and when did you become unable to manage the VAT account pages

Screenshot 2021-06-17 at 12 47 21
Screenshot 2021-06-17 at 12 47 05

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
 technicalIssues.end.error.notInFuture = The date the technology issues ended must be today or in the past
 technicalIssues.end.error.endDateLessThanStartDate = The date the technology issues ended must be the same or after they began
 
+# Was hospital stay required page
+# ----------------------------------------------------------
+healthReason.wasHospitalStayRequired.title = Did this health issue include an unexpected hospital stay?
+healthReason.wasHospitalStayRequired.heading = Did this health issue include an unexpected hospital stay?
+healthReason.wasHospitalStayRequired.yes = Yes
+healthReason.wasHospitalStayRequired.no = No
+healthReason.wasHospitalStayRequired.error.required = Tell us if the health issue included a hospital stay
```
Comment (LewisMitchell) [inline conf/messages:171]: This needs to be `Tell us if you or someone else was admitted to hospital unexpectedly`

File: conf/messages [hunk 2]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+
+# When did the health issue happen page
+# ----------------------------------------------------------
+healthReason.title = When did you become unable to manage the VAT account?
+healthReason.heading = When did you become unable to manage the VAT account?
+healthReason.hintText = Tell us the date when the health issue started to affect your ability to complete this submission.<br><br>For example, 12 3 2018
+healthReason.error.required.all = Enter the date of the crime
```
Comment (LewisMitchell) [inline conf/messages:178]: This needs to be `Enter the start date of the hospital stay`

File: conf/messages [hunk 3]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+healthReason.error.required.two = The date of the crime must include a {0} and a {1}
+healthReason.error.required = The date of the crime must include a {0}
+healthReason.error.invalid = The date of the crime must be a real date
+healthReason.error.notInFuture = The date of the crime must be today or in the past
```
Comment (LewisMitchell) [inline conf/messages:182]: This needs to be `The date the hospital stay started must be today or in the past`

File: conf/messages [hunk 4]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+healthReason.wasHospitalStayRequired.error.required = Tell us if the health issue included a hospital stay
+
+# When did the health issue happen page
+# ----------------------------------------------------------
+healthReason.title = When did you become unable to manage the VAT account?
+healthReason.heading = When did you become unable to manage the VAT account?
+healthReason.hintText = Tell us the date when the health issue started to affect your ability to complete this submission.<br><br>For example, 12 3 2018
+healthReason.error.required.all = Enter the date of the crime
+healthReason.error.required.two = The date of the crime must include a {0} and a {1}
+healthReason.error.required = The date of the crime must include a {0}
```
Comment (LewisMitchell) [inline conf/messages:180]: This needs to be `The date the hospital stay started must include a {0}`

File: conf/messages [hunk 5]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+healthReason.wasHospitalStayRequired.no = No
+healthReason.wasHospitalStayRequired.error.required = Tell us if the health issue included a hospital stay
+
+# When did the health issue happen page
+# ----------------------------------------------------------
+healthReason.title = When did you become unable to manage the VAT account?
+healthReason.heading = When did you become unable to manage the VAT account?
+healthReason.hintText = Tell us the date when the health issue started to affect your ability to complete this submission.<br><br>For example, 12 3 2018
+healthReason.error.required.all = Enter the date of the crime
+healthReason.error.required.two = The date of the crime must include a {0} and a {1}
```
Comment (LewisMitchell) [inline conf/messages:179]: This needs to be `The date the hospital stay started must include a {0} and a {1}`

File: conf/messages [hunk 6]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+healthReason.error.required = The date of the crime must include a {0}
+healthReason.error.invalid = The date of the crime must be a real date
```
Comment (LewisMitchell) [inline conf/messages:181]: This needs to be `The date the hospital stay started must be a real date`

File: conf/messages.cy [hunk 7]
```diff
@@ -162,6 +162,25 @@ technicalIssues.end.error.invalid = The date the technology issues ended must be
...
+# When did the health issue happen page
+# ----------------------------------------------------------
+healthReason.title = When did you become unable to manage the VAT account?
+healthReason.heading = When did you become unable to manage the VAT account?
+healthReason.hintText = Tell us the date when the health issue started to affect your ability to complete this submission.<br><br>For example, 12 3 2018
+healthReason.error.required.all = Enter the date of the crime
+healthReason.error.required.two = The date of the crime must include a {0} and a {1}
+healthReason.error.required = The date of the crime must include a {0}
+healthReason.error.invalid = The date of the crime must be a real date
+healthReason.error.notInFuture = The date of the crime must be today or in the past
```
Comment (LewisMitchell) [inline conf/messages.cy:182]: These will need to be updated with the message changes in the English file

File: it/controllers/HealthReasonControllerISpec.scala [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+            |""".stripMargin)
+      )
+      val request = await(controller.onSubmitForWhenHealthReasonHappened(NormalMode)(fakeRequestWithCorrectKeysAndCorrectBody))
+      request.header.status shouldBe Status.SEE_OTHER
+      // TODO - change to CYA page when navigation added in PRM-329
+      request.header.headers("Location") shouldBe "#"
+      request.session(fakeRequestWithCorrectKeysAndCorrectBody).get(SessionKeys.whenHealthIssueHappened).get shouldBe LocalDate.parse("2021-02-08").toString
+    }
+
+    "return 303 (SEE_OTHER) to CYA page when appeal IS late and add the session key to the session when the body is correct" in {
```
Comment (LewisMitchell) [inline it/controllers/HealthReasonControllerISpec.scala:239]: Should this be `return 303 (SEE_OTHER) to making a late appeal page`?

File: app/views/reasonableExcuseJourneys/health/WasHospitalStayRequiredPage.scala.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+@(form: Form[_], radioOptions: Seq[RadioItem], postAction: Call)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@layout(pageTitle = messages("healthReason.wasHospitalStayRequired.title"),
+    isPageFullWidth = false,
+    formHasErrors = form.errors.nonEmpty,
+    formErrors = form.errors) {
+    @formHelper(action = postAction) {
+        @radios(
+            form = form,
+            legend = messages("healthReason.wasHospitalStayRequired.heading"),
```
Comment (LewisMitchell) [inline app/views/reasonableExcuseJourneys/health/WasHospitalStayRequiredPage.scala.html:18]: The text is slightly bigger on ours versus the prototype - could you add this? 
`legendClasses = Some("govuk-fieldset__legend govuk-fieldset__legend--l"),`

File: test/controllers/HealthReasonControllerSpec.scala [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+  "HealthReasonController" should {
+    "onPageLoadForWasHospitalStayRequired" when {
+
+      "the user is authorised" must {
+        "return OK and correct view" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onPageLoadForWasHospitalStayRequired(NormalMode)(userRequestWithCorrectKeys)
+          status(result) shouldBe OK
+        }
+
+        "return OK and correct view (pre-populated date when present in session) - when answer is no" in new Setup(AuthTestModels.successfulAuthResult) {
```
Comment (LewisMitchell) [inline test/controllers/HealthReasonControllerSpec.scala:67]: Should this be `pre-populated option when present in session`?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some changes needed 🚀

File: app/views/reasonableExcuseJourneys/health/WasHospitalStayRequiredPage.scala.html (hunk 9 above)
Comment (BackRowOla) [reply app/views/reasonableExcuseJourneys/health/WasHospitalStayRequiredPage.scala.html:18]: Same issue with crime, just needed to change the govuk-label in the radio component from xl to l

Comment (LewisMitchell) [reply app/views/reasonableExcuseJourneys/health/WasHospitalStayRequiredPage.scala.html:18]: Oh yes! Good spot!

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**